### ISSUE TITLE
Feature Request: API functionality 

### ISSUE THREAD

**Issue by igorek24** [WillyXJ/facileManager#327]
I want to create a script that would check my home public IP and update fm_dns_records table , but I would still have to log in and reload the zone manually. It would be nice to just call an API.

**Comment by WillyXJ** [comment]
There are security implications that would need to be considered for that to work since fM is based on granular permissions.

**Comment by internethering** [comment]
API would be really nice for automatic renewing lets encrypt certificates when you use dns challange. atm I've to do this manual:

```
certbot certonly --rsa-key-size 4096 --manual --preferred-challenges dns --keep-until-expiring --agree-tos --manual-public-ip-logging-ok --expand -d example.org -d *.example.org

Saving debug log to /var/log/letsencrypt/letsencrypt.log
Plugins selected: Authenticator manual, Installer None
Cert is due for renewal, auto-renewing...
Renewing an existing certificate
Performing the following challenges:
dns-01 challenge for example.org
dns-01 challenge for example.org

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Please deploy a DNS TXT record under the name
_acme-challenge.evilized.de with the following value:

C8WIlKw2Y1z-BgdwsUTrtGG6x5ezluowmr8WVgW_8pw

Before continuing, verify the record is deployed.
```

The possibility to reload the zone per api would be enough for the moment, I would write a script to manipulate fm_dns_records table too. see also https://certbot.eff.org/docs/using.html?highlight=renew#pre-and-post-validation-hooks

**Comment by internethering** [comment]
any news? I've to migrate back to bind only otherwise ...

**Comment by WillyXJ** [comment]
It’s in progress, but development is slow as it’s not my full-time job. Maybe I can get a pre-release version out sooner.

**Comment by WillyXJ** [comment]
This is now implemented in [fmDNS 4.0-beta1](http://www.facilemanager.com/modules/fmdns/?t=3) and later. Testing is requested. Final release will come out later.

**Comment by WillyXJ** [comment]
[fmDNS 4.0.0](http://www.facilemanager.com/modules/fmdns/) has been released.
